flang/runtime/CMakeLists.txt

```diff
@@ -58,3 +57,7 @@ add_flang_library(FortranRuntime
   LINK_LIBS
   FortranDecimal
 )
+
+add_flang_library(Fortran_main STATIC PARTIAL_SOURCES_INTENDED 
+  Fortran_main.c
+)
```

Comment:
LINK_LIBS will add FortranRuntime to the link command when the library is built as a shared library. 
These are the changes I have had to do in the PR with some other libraries to re-enable shared libraries. 
On static libraries it has no effect. Since we always mean for the main to be static, we do not need to specify dependencies.

Reply:
Thanks for explaining.